andela/Ah-backend-xmen
# #162725255  Receive email on successful signup

#### What does this PR do?
implement receive email on successful signup

#### Description of Task to be completed?
- send an email to  a user's email they used on signup 
- verify the email once a user clicks on the verification link sent to their inbox

#### How should this be manually tested?
- fetch this branch by running the command git fetch origin `ft-registration-emial-verification-162725255` in the terminal.
- check into the branch `git checkout ft-registration-emial-verification-162725255`
- add **EMAIL_HOST_USER** and **EMAIL_HOST_PASSWORD** to your .env file
- start the virtual environment `source venv/bin/activate`
- migrate to update database  using `python manage.py migrate`
-  run development server using `python manage.py runserver`
- using postman perform a post request on `http://localhost:8000/api/users/`  to register a user
- provide data in format
        { 
           "user":{
    
        "password":"randompassworD1",
        "email":"radomemail@gmail.com",
        "username":"deoaggrey"
       }     } 
- ensure to enter a valid email 
- check the email you used to signedup with for a link in the format `localhost:8000/api/users/registration-confirmation/MzQ/52z-49bc0b91cdc08b29f3ec/`
- perform a get request on the link and a response will be returned to confirm successful email verification

- run `coverage run --omit="*tests*,*init*,*migrations*,*wsgi*" --source="authors" manage.py test&&coverage report -m` in terminal to test locally

#### What are the relevant pivotal tracker stories?
[#162725255](https://www.pivotaltracker.com/story/show/162725255)

### Screenshots
<img width="759" alt="screen shot 2019-01-14 at 2 41 56 pm" src="https://user-images.githubusercontent.com/39124174/51110828-fc1cdb80-180a-11e9-92a6-bd05c4ec5172.png">
